tests/integration/blackbox_test.py

```diff
@@ -380,6 +408,104 @@ def test_app_with_report(filebase: str):
         invariant.validates(dr_property, dryrun_results)
 
 
+def test_app_itxn_with_report():
+    scenario_success = {
+        "inputs": [()],
+        "invariants": {
+            DRProp.cost: -687,
+            DRProp.budgetConsumed: 13,
+            DRProp.budgetAdded: 700,
+            DRProp.status: "PASS",
+            DRProp.passed: True,
+            DRProp.rejected: False,
+            DRProp.errorMessage: None,
+        },
+    }
+
+    mode = ExecutionMode.Application
+
+    # 0. Validate that the scenario is well defined:
+    inputs, invariants = Invariant.inputs_and_invariants(
+        scenario_success, mode, raw_predicates=True  # type: ignore
+    )
+
+    algod = get_algod()
+
+    # 1. Read the TEAL from ./tests/teal/*.teal
+    path = TESTS_DIR / "teal"
+    case_name = "app_itxn"
+    tealpath = path / f"{case_name}.teal"
+    with open(tealpath, "r") as f:
+        teal = f.read()
+
+    # 2. Run the requests to obtain sequence of Dryrun responses:
+    accounts = [
+        Account(
+            address=get_application_address(Executor.EXISTING_APP_CALL),
+            status="Online",
+            amount=105000000,
+            amount_without_pending_rewards=10500000,
+        )
+    ]
+    dryrun_results = Executor.dryrun_app_on_sequence(algod, teal, inputs, dryrun_accounts=accounts)  # type: ignore
+
+    # 3. Generate statistical report of all the runs:
+    csvpath = path / f"{case_name}.csv"
+    with open(csvpath, "w") as f:
+        f.write(Inspector.csv_report(inputs, dryrun_results))
+
+    print(f"Saved Dry Run CSV report to {csvpath}")
+
+    # 4. Sequential invariants (if provided any)
+    for i, type_n_invariant in enumerate(invariants.items()):
+        dr_property, invariant = type_n_invariant
+
+        assert mode_has_property(
+            mode, dr_property
+        ), f"assert_type {dr_property} is not applicable for {mode}. Please REMOVE or MODIFY"
+
+        invariant = Invariant(invariant, name=f"{case_name}[{i}]@{mode}-{dr_property}")
+        print(
+            f"{i+1}. Semantic invariant for {case_name}-{mode}: {dr_property} <<{invariant}>>"
+        )
+        invariant.validates(dr_property, dryrun_results)
+
+    # test same program without providing app account balance
+    scenario_failure = {
+        "inputs": [()],
+        "invariants": {
+            DRProp.status: "REJECT",
+            DRProp.passed: False,
+            DRProp.rejected: True,
```

Comment:
Is there a particular error message that could be asserted using `DRProp.errorMessage` ? eg: https://github.com/algoidurovic/graviton/blob/7f9146464c451612ed36599f3c718339a0a3ab6c/tests/integration/blackbox_test.py#L640

Reply:
resolving as adding an error message assertion